task_14/Range.cs

```diff
@@ -0,0 +1,91 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace RangeStruct
+{
+    public struct Range
+    {
+        const int MaxValue = int.MaxValue;
+
+        public int Value { get; set; }
+
+        int leftRange;
+        public int LeftRange
+        {
+            get => leftRange;
+            set
+            {
+                if (value > MaxValue)
+                    throw new ArgumentException("Значение должно быть меньше максимального значения целого числа");
+                leftRange = value;
+            }
+        }
+
+        int rightRange;
+        public int RightRange
+        {
+            get => rightRange;
+            set
+            {
+                if ((value > MaxValue && value > leftRange))
+                    throw new ArgumentException("Значение должно быть меньше максимального значения целого числа и больше левой границы");
+                rightRange = value;
```

Comment:
Тип `value` и так будет `int`, так что за пределы диапазона типа не выйдет. Неравенство странное, правая граница должна быть больiе левой. Не предусмотрен имеющийся в задании пустой деапазон
```suggestion
                if ((value < leftRange))
                    throw new ArgumentException("Значение должно быть меньше максимального значения целого числа и больше левой границы");
                else if (value == leftRange)
                {
                    leftRange = 0;
                    rightRange = 0;
                }
                
                rightRange = value;
```